### PR TITLE
Implement StoredKey

### PR DESCRIPTION
**Pull request by alejandro-isaza**
This is part of the Keystore. This first PR adds support for loading and saving encrypted keys with accounts.

### CODE REVIEW

File: src/Address.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#include "Ripple/Address.h"
+#include "Tendermint/Address.h"
+#include "Zcash/TAddress.h"
+
+#include <TrustWalletCore/TWHRP.h>
+
+using namespace TW;
+
+std::string TW::loadAddress(const Data& data, TWCoinType coin) {
+    switch (coin) {
```
Comment (vikmeup) [inline src/Address.cpp]: we don't have concept of blockchain type yet? that would be easier

File: src/Address.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    case TWCoinTypeTron:
+        return Bitcoin::Address(data).string();
+
+    case TWCoinTypeZcash:
+        return Zcash::TAddress(data).string();
+
+    case TWCoinTypeCallisto:
+    case TWCoinTypeDash:
+    case TWCoinTypeEOS:
```
Comment (vikmeup) [inline src/Address.cpp]: should we put this into correct case too? `TWCoinTypeCallisto ` should be under Ethereum and so on

File: src/Keystore/StoredKey.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    static const auto coin = "coin";
+}
+
+namespace UppercaseCodingKeys {
+    static const auto crypto = "Crypto";
+}
+
+namespace TypeString {
+    static const auto privateKey = "private-key";
+    static const auto mnemonic = "mnemonic";
```
Comment (vikmeup) [inline src/Keystore/StoredKey.cpp]: should we consider storing `watch` or `address` types too? as just an address

File: src/Address.cpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    case TWCoinTypeEthereum:
+    case TWCoinTypeEthereumClassic:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+        return Ethereum::Address(data).string();
+
+    case TWCoinTypeICON:
+        return Icon::Address(data, TWIconAddressTypeAddress).string();
+
```
Comment (vikmeup) [inline src/Address.cpp]: `Icon::Address(data, TWIconAddressTypeAddress).string();`
`Bitcoin::Bech32Address(HRP_LITECOIN, 0, data).encode();`

is it possible to make it consistent? so it's always `string()` ?

File: tests/Keystore/Data/legacy-mnemonic.json [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                    {
+                        "value": 0,
+                        "hardened": false
+                    }
+                ],
+                "indexCount": 5
+            }
+        },
+        {
+            "extendedPublicKey": "zpub6r97AegwVxVbJeuDAWP5KQgX5y4Q6KyFUrsFQRn8yzSXrnmpwg1ZKHSWwECR1Kiqgr4h93WN5kdS48KC6hVFniuZHqVFXjULZZkCwurqyPn",
```
Comment (vikmeup) [inline tests/Keystore/Data/legacy-mnemonic.json]: @alejandro-isaza can you make sure `extendedPublicKey` always presented for bitcoin like chains

File: src/Address.cpp (hunk 2 above)
Comment (alejandro-isaza) [reply src/Address.cpp]: I didn't know where they should go, if you do please share :)

File: src/Keystore/StoredKey.cpp (hunk 3 above)
Comment (alejandro-isaza) [reply src/Keystore/StoredKey.cpp]: ok

File: src/Address.cpp (hunk 4 above)
Comment (alejandro-isaza) [reply src/Address.cpp]: ok

File: src/Address.cpp (hunk 2 above)
Comment (vikmeup) [reply src/Address.cpp]: `ethereum` like: `TWCoinTypeCallisto`, `TWCoinTypePoa`, `TWCoinTypeGo`, `TWCoinTypeThunderToken`, `TWCoinTypeTomoChain`
`bitcoin like: `TWCoinTypeDash`, `TWCoinTypeZcoin`
`TWCoinTypeTezos ` - this should go to `TezosAddres()`
`TWCoinTypeEOS` - not implemented

Comment (hewigovens) [reply src/Address.cpp]: Go/Poa/ThunderToken/TomoChain are eth like, Zcoin is Bitcoin like

File: src/DerivationPath.h [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+    void setAddress(uint32_t v) {
+        indices[4] = DerivationPathIndex(v, /* hardened: */false);
+    }
+
+    DerivationPath() = default;
+    DerivationPath(std::vector<DerivationPathIndex> indices) : indices(indices) {}
+
+    /// Creates a `DerivationPath` by components.
+    DerivationPath(TWPurpose purpose, TWCoinType coin, uint32_t account = 0, uint32_t change = 0, uint32_t address = 0) {
```
Comment (hewigovens) [inline src/DerivationPath.h]: Can we support different number of levels? e.g. for Stellar (ed25519) path is `m/44'/148'/0'`

Comment (alejandro-isaza) [reply src/DerivationPath.h]: done

File: tests/Keystore/Data/legacy-mnemonic.json (hunk 5 above)
Comment (alejandro-isaza) [reply tests/Keystore/Data/legacy-mnemonic.json]: done

**Comment by hewigovens** [comment]
We will have Swift / Java bindings eventually?

File: src/Address.cpp (hunk 2 above)
Comment (alejandro-isaza) [reply src/Address.cpp]: Done

**Comment by alejandro-isaza** [comment]
Yes, the interface is coming next along with Keystore.

**Pull request merged**